### ISSUE TITLE
SPFx error during `gulp serve`

### ISSUE THREAD

**Issue by DonKirkham** [SharePoint/sp-dev-docs#8930]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Linux

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version: I use Edge Version 114.0.1807.1 (Official build) dev (64-bit), but error prevents browser from opening
- SPFx 1.17.1
- Node.js 16.19.1
- WSL2
- Ubuntu 22.04 LTS

I reproduced the error on multiple machines.  I did not attempt it on SPFx v1.17.0, but all SPFx versions < 1.17.x do not have this issue.

### Describe the bug / error

NOTE: "mydevtenant" represents my actual tenant domain in all entries below
In OOTB HelloWorld webpart, the following error happens during `gulp serve`
```
❯ gulp serve
Build target: DEBUG
[07:54:48] Using gulpfile ~/..code/play/latest2/gulpfile.js
[07:54:48] Starting 'serve'...
[07:54:48] Starting gulp
[07:54:48] Starting subtask 'spfx-serve'...
[07:54:48] [spfx-serve] To load your scripts, use this query string: ?debug=true&noredir=true&debugManifestsFile=https://localhost:4321/temp/manifests.js
[07:54:48] Starting server...
[07:54:48] Finished subtask 'spfx-serve' after 77 ms
[07:54:48] Starting subtask 'pre-copy'...
[07:54:48] Finished subtask 'pre-copy' after 15 ms
[07:54:48] Starting subtask 'copy-static-assets'...
[07:54:48] Starting subtask 'sass'...
[07:54:48] Server started https://localhost:4321
[07:54:48] LiveReload started on port 35729
[07:54:48] Running server
[07:54:48] Opening https://mydevtenant.sharepoint.com/_layouts/workbench.aspx using the default OS app
[07:54:48] Finished subtask 'sass' after 313 ms
[07:54:48] Starting subtask 'lint'...
[07:54:48] [lint] eslint version: 8.7.0
[07:54:48] Starting subtask 'tsc'...
[07:54:48] [tsc] typescript version: 4.5.5
  Request: '/sockjs-node/info?t=1682686488020'
[07:54:50] Finished subtask 'copy-static-assets' after 1.9 s
[07:54:50] 'serve' errored after 2.73 s
[07:54:50] Error: Exited with code 3
    at ChildProcess.<anonymous> (/home/don/..code/play/latest2/node_modules/opn/index.js:84:13)
    at Object.onceWrapper (node:events:628:26)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess.emit (node:domain:552:15)
    at maybeClose (node:internal/child_process:1100:16)
    at Socket.<anonymous> (node:internal/child_process:458:11)
    at Socket.emit (node:events:513:28)
    at Socket.emit (node:domain:552:15)
    at Pipe.<anonymous> (node:net:301:12)
    at Pipe.callbackTrampoline (node:internal/async_hooks:130:17)
About to exit with code: 1
Process terminated before summary could be written, possible error in async code not continuing!
Trying to exit with exit code 1
```
- The browser does not open
- Running `gulp serve --nobrowser` does not produce the error.
- 'mydevtenant.sharepoint.com` is my tenant domain, set in the environment variable SPFX_SERVE_TENANT_DOMAN
```
❯ echo $SPFX_SERVE_TENANT_DOMAIN
mydevtenant.sharepoint.com
```
- If I replace {tenantDomain} with my tenant domain directly in the serve.json file, the error is the same
- I built the HelloWorld webpart solution 3 times. Only change was framework. I tried all 3: minimal, no framework, and React. All have the same issue.

### Steps to reproduce

1. Set up dev environment as follows
```
❯ npm ls -g
/home/don/.nvm/versions/node/v16.19.1/lib
├── @microsoft/generator-sharepoint@1.17.1
├── corepack@0.15.1
├── gulp-cli@2.3.0
├── npm@8.19.3
└── yo@4.3.1
```
2. Set SPFX_SERVE_TENANT_DOMAIN environment variable
```
❯ export SPFX_SERVE_TENANT_DOMAIN=mydevtenant.sharepoint.com
```
4. Create new webpart solution with `yo @microsoft/sharepoint`
5. Select default options, any framework
6. After creation and dependency installation, run `gulp serve`

### Expected behavior

Expectation is that `gulp serve` runs without errors and browser opens to the workbench site listed in the "Opening ..." line.

**Comment by AJIXuMuK** [comment]
@DonKirkham - thanks for reporting the issue.
I can repro it on WSL.
Will be investigating further

**Comment by DonKirkham** [comment]
The {tenantDomain} placeholder in the debugging profile (launch.json) also does not work. The page that is opened is https://%7Btenantdomain%7D/_layouts/workbench.aspx

The default type of "edge" is also incorrect, but I turned that in as a separate issue. #8936

**Comment by AJIXuMuK** [comment]
The `{tenantDomain}` is replaced with OS variable if that exists.
Otherwise yes, it will just open the URL you mentioned above.

**Comment by DonKirkham** [comment]
The only environment variable I know of is SPFX_SERVE_TENANT_DOMAIN. It is set as shown below:
```
❯ echo $SPFX_SERVE_TENANT_DOMAIN
mydevtenant.sharepoint.com
```

**Comment by AJIXuMuK** [comment]
Sorry @DonKirkham - I'm a bit confused now.
Were you able to make `gulp serve` work in WSL2?
Your initial issue is the serve task is failing completely.
But now you're saying the problem is with incorrect URL in the browser.

**Comment by DonKirkham** [comment]
Sorry for the confusion.  The original reported error is still occurring unless you use the --nobrowser option.

I have made an assumption that the original error is related to the {tenantDomain} placeholder, but I could be badly mistaken (again) since replacing the {tenantDomain} placeholder in the serve.json file with a real url results in the same error.

The {tenantDomain} issue in launch.json might be unrelated since manually replacing it with a real url results in no errors during debug. (If you used --nobrowser to in the gulp serve command.

These may be two unrelated issues, but since I think they have something to do with adding the {tenantDomain} placeholder functionality, I added the second part here.  I can open a separate issue, if needed.

**Comment by AJIXuMuK** [comment]
@DonKirkham - good new and bad news :)
Bad news: it didn't work in previous versions of SPFx too as the issue is not related to SPFx directly.
We're using `gulp-open` (yes, I know it's old) which indirectly uses `is-wsl` of version. And it returns `false` for WSL2.

Now good news is: https://github.com/microsoft/rushstack-legacy/pull/24

When this one is merged we will update SPFx build routine and release new version.

**Comment by AJIXuMuK** [comment]
fixed in 1.17.2

**Comment by DonKirkham** [comment]
tested and confirmed fixed!  Thanks Alex
